### PR TITLE
fix: the quickstart guide needs update for external cloud provider

### PR DESCRIPTION
**Pull request by Skarlso**
**What type of PR is this?**
/kind bug
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4299

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
The quick start guide is generated using this template and needs to use external provider.
```

### CODE REVIEW

**Comment by Skarlso** [comment]
Uh, I'm not sure about all of this. There is something not okay. There are templates in our e2e folder that use aws provider and still work. Yet this thing doesn't. I'm not sure what's up.

File: test/e2e/data/infrastructure-aws/withclusterclass/e2e_test_templates/cluster-template-external-vpc-clusterclass.yaml [hunk 1]
```diff
@@ -0,0 +1,1078 @@
+apiVersion: cluster.x-k8s.io/v1beta1
```
Comment (Ankitasw) [inline test/e2e/data/infrastructure-aws/withclusterclass/e2e_test_templates/cluster-template-external-vpc-clusterclass.yaml:1]: This file is not required in templates, as we are not running this test.

**Comment by richardcase** [comment]
Just to test the EKS prow change:

/test pull-cluster-api-provider-aws-e2e-eks-gc

File: test/e2e/data/infrastructure-aws/withclusterclass/e2e_test_templates/cluster-template-external-vpc-clusterclass.yaml (hunk 1 above)
Comment (Skarlso) [reply test/e2e/data/infrastructure-aws/withclusterclass/e2e_test_templates/cluster-template-external-vpc-clusterclass.yaml:1]: Will remove. Thanks.

**Pull request merged**